openrtb_ext/user.go

```diff
@@ -2,6 +2,11 @@ package openrtb_ext
 
 // ExtUser defines the contract for bidrequest.user.ext
 type ExtUser struct {
+
+	// Consent is a GDPR consent string. See "Advised Extensions" of
+	// https://iabtechlab.com/wp-content/uploads/2018/02/OpenRTB_Advisory_GDPR_2018-02.pdf
+	Consent string `json:"consent"`
```

Comment:
Just to confirm, there is a use-case where the lookup process to grab a person's GDPR consent ID string could fail and the publisher would still want the auction to proceed without the ID.

In this case, we're going to set the consent ID to `undefined` and pass it along.  Would the code as is support this type of value?

Reply:
Let's chat about this tomorrow. I need to get a better feel for the use-cases.

As written, Go will fill this out with an empty string if you pass it undefined. It'll pass this onto each Bidder, and they can do whatever they want with it.

I haven't done the GDPR research... so I need to understand what "failed lookup" means, legally & semantically.

Reply:
Won't it pass the original `user.ext` to the bidder, which will leave it undefined as submitted? I don't see where we are rewriting it here, and can't think of why we would rewrite it elsewhere.

Reply:
I think he means the javascript `undefined`, and not the string `"undefined"`.

JSON doesn't have an `undefined`, so I _think_ the browser would omit that key in the object it sends. Go would `Unmarshal` that as the zero object, which is an empty string.

Not sure, though... let's iron out the details tomorrow.